### PR TITLE
isSelected and common parents

### PR DESCRIPTION
**Pull request by TAImatem**
added IsSelected to ModInput and ModButton
added common parents where it made sense (Not sure if I should also implement class ModInputBase, common interface sounds enough for storage purposes)

### CODE REVIEW

**Comment by TAImatem** [comment]
IsSelected is tested only for ModInputCombinationElement

**Comment by TAImatem** [comment]
Although it kinda implies it would also work for all TwoButtonToggle-based things
Not yet sure if UIStyleApplier-based IsSelected works flawlessly

File: OWML.ModHelper.Menus/ModButton.cs [hunk 1]
```diff
@@ -22,6 +23,17 @@ public int Index
                 Button.transform.SetSiblingIndex(value);
             }
         }
+        public bool IsSelected => _uIStyleApplier.GetValue<bool>("_selected");
```
Comment (amazingalek) [inline OWML.ModHelper.Menus/ModButton.cs:26]: can _uIStyleApplier be null? If so, this'll probably do:
`_uIStyleApplier?.GetValue<bool>("_selected") ?? false;`

File: OWML.Common/Menus/IModInput.cs [hunk 2]
```diff
@@ -3,15 +3,9 @@
 
 namespace OWML.Common.Menus
 {
-    public interface IModInput<T>
+    public interface IModInput<T> : IModInputBase
```
Comment (amazingalek) [inline OWML.Common/Menus/IModInput.cs:6]: This seems to be the only place that inherits from IModInputBase, so why should IModInput and IModInputBase be split?

**Comment by amazingalek** [comment]
Super cool stuff! :)
Not sure we have the best naming of all of these UI elements tho ;)
Example: ModMenuInput, ModInputField - can we find more descriptive names?

File: OWML.ModHelper.Menus/ModButton.cs (hunk 1 above)
Comment (TAImatem) [reply OWML.ModHelper.Menus/ModButton.cs:26]: ah, didn't think to use `??`

File: OWML.Common/Menus/IModInput.cs (hunk 2 above)
Comment (TAImatem) [reply OWML.Common/Menus/IModInput.cs:6]: because IModInput<bool> != IModInput<string>

File: OWML.ModHelper.Menus/ModButton.cs (hunk 1 above)
Comment (TAImatem) [reply OWML.ModHelper.Menus/ModButton.cs:26]: Wait for sometime though I want to implement ModPromptButton in a separate branch first

Comment (TAImatem) [reply OWML.ModHelper.Menus/ModButton.cs]: done

File: OWML.Common/Menus/IModInput.cs (hunk 2 above)
Comment (TAImatem) [reply OWML.Common/Menus/IModInput.cs:6]: Idea is that you can do `Container<IModInputBase>`, but can't `Container<IModInput>`

**Comment by TAImatem** [comment]
maybe ModFieldInput
Also ModPopupInput intead of ModMenuInput might be better

**Comment by amazingalek** [comment]
Awesome.
You have a couple unused usings in interfaces (IModInput and IModInputBase)

**Pull request merged**